src/ui/public/vis/request_handlers/courier.js

```diff
@@ -72,22 +73,11 @@ const CourierRequestHandlerProvider = function (Private, courier, timefilter) {
         searchSource.set('query', appState.query);
       }
 
-      // AggConfig contains circular reference to vis, which contains visualization parameters,
-      // which we should not look at
-      const copyAggs = (aggs) => {
-        return aggs.map(agg => {
-          return {
-            type: agg.type,
-            params: agg.params
-          };
-        });
-      };
-
       const shouldQuery = () => {
         if (!searchSource.lastQuery || vis.reload) return true;
         if (!_.isEqual(_.cloneDeep(searchSource.get('filter')), searchSource.lastQuery.filter)) return true;
```

Comment:
Maybe outside this PR, but is there any reason not to use calculateObjectHash instead of cloneDeep for all of these comparisons?

Reply:
Yeah we already discussed that, we will touch this topic, but not for fixing that blocker bug. Instead we'll redefine the object hash interface properly and use it here for all the props and do some performance comparison on those.